network/src/noise/handshake.rs

```diff
@@ -304,6 +304,19 @@ impl NoiseUpgrader {
             )
         })?;
 
+        // prevent accidental self-dials
+        // this situation could occur either as a result of our own discovery
+        // mis-configuration or a potentially malicious discovery peer advertising
+        // a (loopback ip or mirror proxy) and our public key.
+        if remote_peer_id == self.self_peer_id {
```

Comment:
do you want to update our network-noise spec with this change :) ? should be a one liner